crypto/poseidon/codehash.go

```diff
@@ -4,44 +4,37 @@ import (
 	"math/big"
 
 	"github.com/scroll-tech/go-ethereum/common"
-	"github.com/scroll-tech/go-ethereum/crypto"
 )
 
 const defaultPoseidonChunk = 3
+const nBytesToFieldElement = 31
 
 func CodeHash(code []byte) (h common.Hash) {
-	// special case for nil hash
-	if len(code) == 0 {
-		return crypto.Keccak256Hash(nil)
-	}
-
-	cap := int64(len(code))
+	nBytes := int64(len(code))
 
-	// step 1: pad code with 0x0 (STOP) so that len(code) % 16 == 0
-	// step 2: for every 16 bytes, convert to Fr, so that we get a Fr array
-	var length = (len(code) + 15) / 16
+	// step 1: pad code with 0x0 (STOP) so that len(code) % nBytesToFieldElement == 0
+	// step 2: for every nBytesToFieldElement bytes, convert to Fr, so that we get a Fr array
+	var length = (len(code) + nBytesToFieldElement - 1) / nBytesToFieldElement
```

Comment:
@noel2004 @Thegaram Could you help review the new codehash computation logic?

Reply:
The comutation logic is LGTM